host/src/gap.rs

```diff
@@ -75,7 +81,7 @@ pub struct PeripheralConfig<'a> {
 /// Configuration for a central device GAP Service.
 pub struct CentralConfig<'a> {
     /// The name of the central device.
-    pub name: &'a [u8; 22],
+    pub name: &'a Vec<u8, DEVICE_NAME_MAX_LENGTH>,
```

Comment:
Bluetooth specifies the name to be UTF-8 so I think this should be a `String` instead. And if it is taken by reference it should be a `str` instead of a `String`.

It would also be fine to take it by value as `name: String<DEVICE_NAME_MAX_LENGTH>`, which would eliminate the lifetime parameter. What we can't do is take a `'static` reference like: `name: &'static str` because that makes it difficult to use runtime-determined values for the name.

Reply:
Yes this and the static cell below was the result of me running into a sticky situation with  not being able to implement `GattValue` on `&str` because a) it seems to implicitly dereference to `str` which is not `Sized` and b) the lifetime of the associated `&[u8]` gets tied to the lifetime of the `&str`, which according to the compiler doesn't conform to the trait signature.
This code isn't compatible with the API as we want it, so this will definitely be changing.

To get around problem a) I'm going to try implementing a wrapper type around `&str` to see if that avoids the implicit deref.
I think the mem::transmute should take care of problem B) but I need to work on it more.

If you have any ideas about how better to deal with these issues though I'd be very keen to hear them! :)

Reply:
If the config contains references, you'll want to copy the data into owned storage in the server (e.g. the server will have a buffer somewhere (perhaps in the attributes table) that is `DEVICE_NAME_MAX_LENGTH` bytes and on initialization you'll copy the config data into that buffer). If the config contains owned values, you would just. move the value into the attributes table as part of initialization.

I don't think you need the values in the config struct to be `GattValue`s. The type of the device name characteristic in the attribute table should be `heapless::String<DEVICE_NAME_MAX_LENGTH>` which should be straightforward to implement `GattValue` for.